src/com/pb/javatraining/service/StatementService.java

```diff
@@ -15,66 +16,101 @@
+                .flatMap(group -> group.getStudents().stream())
+                .collect(Collectors.toList());
+    }
 
-        return 0;
+    public int getAmountOfGroupsWhereBadStudents() {
+        return (int) groups.stream()
+                .filter(group -> group.getStudents().stream()
+                        .mapToDouble(student -> student.getMarksByLessons().entrySet().stream()
+                                .mapToDouble(Map.Entry::getValue)
+                                .average()
+                                .getAsDouble())
+                        .anyMatch(average -> average < 60))
+                .count();
     }
 
-    public Map<Group, Integer> getAverageMarksWithinGroups() {
 
-        // TODO
 
-        return emptyMap();
+    public Map<Group, Double> getAverageMarksWithinGroups() {
+        return groups.stream().collect(Collectors.toMap(group -> group,
+                group ->  group.getStudents().stream()
+                        .mapToDouble(student -> student.getMarksByLessons().entrySet().stream()
+                                .mapToDouble(Map.Entry::getValue)
+                                .average()
+                                .getAsDouble())
+                        .average()
+                        .getAsDouble()));
     }
 
     public Collection<String> getGroupTitlesWhereStudentsAreMen() {
-
-        // TODO
-
-        return emptyList();
+       return groups.stream()
+                .filter(group -> group.getStudents().stream().
+                        allMatch(student -> student.getState() == State.MALE))
+                .map(Group::getTitle)
+                .collect(Collectors.toList());
     }
 
     public Collection<String> getStudentFullNamesFromSpecifiedBranch(Branch branch) {
-
-        // TODO
-
-        return emptyList();
+        return data.stream()
+                .filter(b -> b.equals(branch))
+                .flatMap(b -> b.getGroups().stream())
+                .flatMap(group -> group.getStudents().stream())
+                .map(student -> student.getName() + " " + student.getSurname())
+                .collect(Collectors.toList());
     }
 
     public Map<Lesson, Double> getAverageMarksWithinLessons() {
-
-        // TODO
-
-        return emptyMap();
+        return getAllLessons().stream()
+                .collect(Collectors.toMap(lesson -> lesson, lesson -> getAllStudents().stream()
+                        .flatMap(student -> student.getMarksByLessons().entrySet().stream())
+                        .filter(l -> l.getKey().equals(lesson))
+                        .mapToDouble(Map.Entry::getValue)
+                        .average()
+                        .getAsDouble()));
     }
 
     public Collection<Student> getStudentsSuitableForArmy() {
-
-        // TODO
-
-        return emptyList();
+        return getAllStudents().stream()
+                .filter(student -> student.getState() == State.MALE && student.getAge()>=18)
```

Comment:
Separate this one on two independent:
`...filter(s -> s.getState() == State.MALE).filter(s -> s.getAge() >= 18)..`